sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobClient.java

```diff
@@ -514,12 +637,51 @@ public Response<BlockBlobItem> uploadWithResponse(InputStream data, long length,
     public Response<BlockBlobItem> uploadWithResponse(BlockBlobSimpleUploadOptions options, Duration timeout,
         Context context) {
         StorageImplUtils.assertNotNull("options", options);
-        Mono<Response<BlockBlobItem>> upload = client.uploadWithResponse(options, context);
+
+        Supplier<Response<BlockBlobItem>> operation = () ->
+            uploadWithResponseSync(options, context);
+
         try {
-            return blockWithOptionalTimeout(upload, timeout);
+            return timeout != null
+                ? THREAD_POOL.submit(() -> operation.get()).get(timeout.toMillis(), TimeUnit.MILLISECONDS)
+                : operation.get();
         } catch (UncheckedIOException e) {
             throw LOGGER.logExceptionAsError(e);
+        } catch (ExecutionException | TimeoutException | InterruptedException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+    }
```

Comment:
Instead of this, can leverage method:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java#L457